File: 1 basic/1020. 月饼 (25).cpp
#include<iostream>
#include<cstdio>
#include<algorithm> 
using namespace std;

struct MoonCake{
	double amount,money;
	double price;
};
bool cmp(MoonCake m1,MoonCake m2){
	return m1.price > m2.price;
}

int main(){
	int N,D;
	scanf("%d %d",&N,&D);
	MoonCake mc[N];
	for(int i=0;i<N;++i){
		scanf("%lf",&mc[i].amount);
	}
	for(int i=0;i<N;++i){
		scanf("%lf",&mc[i].money);
		mc[i].price = mc[i].money/mc[i].amount;
	}
	sort(mc,mc+N,cmp);
	
//	for(int i=0;i<N;++i)
//		printf("%.2lf\n",mc[i].price);
		
	double profit=0;
	for(int idx = 0; D && idx<N; ++idx){
		if(D >= mc[idx].amount){
			profit += mc[idx].money;
			D -= mc[idx].amount;
		}
		else{
			profit += D * mc[idx].price;
			D = 0;
		}
	}

	printf("%.2lf",profit);

	return 0; 
} 

/*
75 72 45 money
18 15 10 amount
4.16 4.8 4.5
*/

File: 1 basic/py/1020. 月饼 (25).py
class MoonCake(object):
    def __init__(self, total, money):
        self.total = total
        self.money = money
        self.price = money / total
    def __str__(self):
        return '%s %s %s '%(self.total,self.money,self.price)

n,need = input().split()
n,need = int(n),int(need)
s = list(map(int, input().split()))
p = list(map(int, input().split()))
mc = []
for i in range(n): mc.append(MoonCake(s[i], p[i]))
mc.sort(key = lambda obj: obj.price, reverse=True)

profit = 0
for o in mc:
    if need >= o.total:
        need = need - o.total
        profit = profit + o.money
    else:
        profit = profit + need * o.price
        break
print('{0:.2f}'.format(profit))